Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -0,0 +1,195 @@
+  have half_ε_pos : 0 < ε / 2 := by
+    simp only [ENNReal.div_pos_iff, ne_eq, ε_pos.ne.symm, not_false_eq_true, two_ne_top, and_self]
+  let r := levyProkhorovEDist μ ν + ε / 2
+  let s := levyProkhorovEDist ν κ + ε / 2
+  have lt_r : levyProkhorovEDist μ ν < r := lt_add_right LPμν_finite half_ε_pos.ne.symm
+  have lt_s : levyProkhorovEDist ν κ < s := lt_add_right LPνκ_finite half_ε_pos.ne.symm
+  refine ⟨?_, ?_⟩
+  · have obs₁ := left_measure_le_of_levyProkhorovEDist_lt lt_r B_mble
+    have mble_B₁ : MeasurableSet (thickening (ENNReal.toReal r) B) :=
+      isOpen_thickening.measurableSet
+    have obs₂ := left_measure_le_of_levyProkhorovEDist_lt lt_s mble_B₁
+    simp only [ENNReal.div_pos_iff, ne_eq, and_true, gt_iff_lt,
+               FiniteMeasure.ennreal_coeFn_eq_coeFn_toMeasure, ge_iff_le] at *
+    apply obs₁.trans
+    apply (add_le_add_right obs₂ _).trans
+    simp_rw [add_assoc, add_comm (ε / 2), add_assoc, ENNReal.add_halves]
+    simp_rw [← add_assoc (levyProkhorovEDist _ _),
+             add_comm (levyProkhorovEDist μ ν) (levyProkhorovEDist ν κ)]
+    congr
+    apply add_le_add_right (measure_mono _)
+    apply (thickening_thickening_subset _ _ _).trans (thickening_mono _ _)
+    rw [← ENNReal.toReal_add]
+    · simp_rw [add_assoc, add_comm (ε / 2), add_assoc, ENNReal.add_halves, ← add_assoc]
+      exact Eq.le rfl
+    · simp only [add_ne_top]
+      refine ⟨ne_top_of_lt lt_s, (ENNReal.div_lt_top ε_lt_top.ne two_ne_zero).ne⟩
+    · simp only [add_ne_top]
+      refine ⟨ne_top_of_lt lt_r, (ENNReal.div_lt_top ε_lt_top.ne two_ne_zero).ne⟩
+  · have obs₁ := right_measure_le_of_levyProkhorovEDist_lt lt_s B_mble
+    have mble_B₁ : MeasurableSet (thickening (ENNReal.toReal s) B) :=
+      isOpen_thickening.measurableSet
+    have obs₂ := right_measure_le_of_levyProkhorovEDist_lt lt_r mble_B₁
+    simp only [ENNReal.div_pos_iff, ne_eq, and_true, gt_iff_lt,
+      FiniteMeasure.ennreal_coeFn_eq_coeFn_toMeasure, ge_iff_le] at *
+    apply obs₁.trans
+    apply (add_le_add_right obs₂ _).trans
+    simp_rw [add_assoc, add_comm (ε / 2), add_assoc, ENNReal.add_halves]
+    simp_rw [← add_assoc (levyProkhorovEDist _ _),
+             add_comm (levyProkhorovEDist μ ν) (levyProkhorovEDist ν κ)]
+    congr
+    apply add_le_add_right (measure_mono _)
+    apply (thickening_thickening_subset _ _ _).trans (thickening_mono _ _)
+    rw [← ENNReal.toReal_add]
+    · simp_rw [add_assoc, add_comm (ε / 2), add_assoc, ENNReal.add_halves, ← add_assoc]
+      rw [add_comm (levyProkhorovEDist _ _)]
+    · simp only [add_ne_top]
+      refine ⟨ne_top_of_lt lt_r, (ENNReal.div_lt_top ε_lt_top.ne two_ne_zero).ne⟩
+    · simp only [add_ne_top]
+      refine ⟨ne_top_of_lt lt_s, (ENNReal.div_lt_top ε_lt_top.ne two_ne_zero).ne⟩
+
+/-- The Lévy-Prokhorov distance `levyProkhorovEDist` makes `FiniteMeasure X` a pseudoemetric
+space. -/
+noncomputable def levyProkhorovEDist_pseudoEMetricSpace_finiteMeasure :
+    PseudoEMetricSpace (FiniteMeasure Ω) where
```

Comment:
With this definition, you are registering a new topology on `FiniteMeasure Ω`. This is not a problem, since this definition is not registered as an instance, but if one day you want to register it as an instance you better make sure that the topology it defines is definitionally the same as the usual topology given by weak convergence against bounded functions. This is not the case currently. Are you planning to refactor the definition once you have shown that the topology generated by the Levy-Prokhorov distance is the weak topology? Or use this one as an intermediate step? In any case, I think this is worth discussing in the docstring. Same thing for the next definition.

Reply:
Yes, I did not make this an instance because the topology of weak convergence is more generally the right one. 

Under the further assumptions of Borel sigma algebra and separability, the generated topology will be the same (propeq, I don't know whether it could be made defeq).

I mainly intended to use the Lévy-Prokhorov metric to show metrizability of the topology of weak convergence under those assumptions; I thought for just metrizability defeqs should not matter. I did not plan to add the (pseudo)metric as a global instance, just add metrizability as an instance under the assumptions that it holds.

Do these make sense? I don't have strong opinions about them, but the point is that my plan with Lévy-Prokhorov metric was very limited.

Reply:
I will add some comment in the docstring.

Reply:
Actually maybe these should be only put on a type synonym as Yury suggested.

Indeed the `PseudoEMetricSpace` would be a kind of an intermediate step when I would show comparisons of the two topologies: the LP-metric one is generally finer than that of weak convergence, but in separable Borel spaces equal. For this comparison of topologies I wanted a `PseudoEMetricSpace` with its topology (but which was not meant as an instance). The type synonym approach seems less confusing anyway, so I will switch to it.